core/domain/activity_jobs_one_off.py

```diff
@@ -375,24 +375,28 @@ def _migrate_user_id(snapshot_model):
             (str, str). Result info, first part is result message, second is
             additional info like IDs.
         """
-        new_user_ids = []
+        # Only commit_cmds of length 2 are processed by this method.
+        new_user_ids = [None, None]
         for i, commit_cmd in enumerate(snapshot_model.commit_cmds):
+            assignee_id = commit_cmd['assignee_id']
             if (
                     commit_cmd['cmd'] == rights_domain.CMD_CHANGE_ROLE and
-                    not user_services.is_user_id_correct(
-                        commit_cmd['assignee_id'])
+                    not user_services.is_user_id_correct(assignee_id)
             ):
-                old_user_id = commit_cmd['assignee_id']
                 user_settings_model = (
-                    user_models.UserSettingsModel.get_by_gae_id(old_user_id))
+                    user_models.UserSettingsModel.get_by_gae_id(assignee_id))
                 if user_settings_model is None:
                     return (
-                        'MIGRATION_FAILURE', (snapshot_model.id, old_user_id))
+                        'MIGRATION_FAILURE', (snapshot_model.id, assignee_id))
 
-                new_user_ids.append(user_settings_model.id)
+                new_user_ids[i] = user_settings_model.id
 
+        # This loop is used for setting the actual commit_cmds and is separate
+        # because if the second model results in MIGRATION_FAILURE we do not
```

Comment:
I don't really understand this. What do you mean by the "second model"?

I feel like way more context is needed here, generally speaking.

Reply:
Okay, this was a mistake, I have updated the comment.